File: c/seperateEvenOdd.c
// Separate even and odd numbers in an Array using C language
#include <stdio.h>
int main(){
    int array[30] = {0};
    int n = 0;
    int even_array[30] = {0};
    int odd_array[30] = {0};
    int even_count = 0, odd_count = 0;
    printf("Enter the number of elements you wanted to insert:  ");
    scanf("%d",&n);
    printf("\n");
    
    printf("Enter each number one by one:   \n");
    for (int i = 0; i < n; i++) {
        scanf("%d",&array[i]);
    }
    
    for(int i=0; i<n;i++){
        if(array[i] % 2 == 0){
           even_array[even_count] = array[i];
           even_count++;
        }
        else{
            odd_array[odd_count] = array[i];
            odd_count++;
        }
    }
    printf("Even Number List:   ");
    for(int i=0; i<even_count; i++){
        printf("%d ",even_array[i]);
    }
    printf("\n");
    
    printf("Odd Number List:   ");
    for(int i=0; i<odd_count; i++){
        printf("%d ",odd_array[i]);
    }
    printf("\n");
    
    
    return 0;
}

File: java/SeperateEvenOdd.java
// Java program to sepeate Even and Odd Numbers
import java.util.Scanner;
public class SeperateEvenOdd{
    public static void main(String[] args){
        int array[] = new int[30];
        int even_array[] = new int[30];
        int odd_array[] = new int[30];
        Scanner sc = new Scanner(System.in);
        int even_count = 0, odd_count = 0;
        int n = 0;
        System.out.println("Enter number of element you wanted to insert:   ");
        n = sc.nextInt();
        
        System.out.println("Enter each element one by one:  ");
        for (int i=0;i<n ;i++ ) {
            array[i] = sc.nextInt();
        }
        
        for(int i=0; i<n; i++){
            if(array[i]%2 == 0){
                even_array[even_count] = array[i];
                even_count++;
            }
            else{
                odd_array[odd_count] = array[i];
                odd_count++;
            }
        }
        System.out.print("Even Elements: ");
        for(int i=0; i<even_count; i++){
            System.out.print(even_array[i]+" ");
        }
        System.out.println("");
        System.out.print("Odd Elements: ");
        for(int i=0; i<odd_count; i++){
            System.out.print(odd_array[i]+" ");
        }
    }
}